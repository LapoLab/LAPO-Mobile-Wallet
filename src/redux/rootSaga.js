import { all } from 'redux-saga/effects';
import commonsSagas from '../commons/sagas';
import recoveryWordsSagas from '../modules/RecoveryWords/sagas';
import sendSagas from '../modules/Send/sagas';
import paperWordsConfirm from '../modules/PaperWordsConfirm/sagas';
import dashboard from '../modules/Dashboard/sagas';
import settings from '../modules/Settings/sagas';
import paperWords from '../modules/PaperWords/sagas';
import transactionsDetails from '../modules/TransactionDetails/sagas';
import chooseContact from '../modules/ChooseContact/sagas';
import splashScreen from '../modules/SplashScreen/sagas';
import switchContacts from '../modules/SwitchContacts/sagas';
import receive from '../modules/Receive/sagas';

// eslint-disable-next-line
const getListeners = (...args) => args.reduce((acc, nextArg) => [...acc, ...Object.values(nextArg).map(func => func())], []);

export default function* rootSaga() {
  yield all(
    getListeners(
      commonsSagas,
      recoveryWordsSagas,
      sendSagas,
      paperWordsConfirm,
      dashboard,
      settings,
      paperWords,
      transactionsDetails,
      chooseContact,
      splashScreen,
      switchContacts,
      receive,
    ),
  );
}
