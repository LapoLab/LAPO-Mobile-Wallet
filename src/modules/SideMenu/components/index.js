// @flow
import React, { Component } from 'react';
import {
  ScrollView,
  Text,
  View,
  Image,
  Dimensions,
  SafeAreaView,
  TouchableOpacity,
} from 'react-native';
import { NavigationActions } from 'react-navigation';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Share from 'react-native-share';
import type { Dispatch } from 'redux';
import {
  setModalSend as setModalSendAction,
  setModalReceive as setModalReceiveAction,
  setPaperWordsConfirm as setPaperWordsConfirmAction,
  setPins as setPinsAction,
  setConfirmPins as setConfirmPinsAction,
  goToOtherWindow as goToOtherWindowAction,
} from '../../../commons/actions';
import * as actionNotify from '../../Notify/actions';

import styles from './styles';
import {
  dashboard, send, recieve, trans, notify, profile, setup,
} from '../../../assets';

const { height: heightWindow } = Dimensions.get('window');

type Props = {
  navigation: {
    dispatch: Function,
    closeDrawer: Function,
  },
  setModalSend: Function,
  setModalReceive: Function,
  setVisibleNotify: Function,
  username: string,
  avatarSource: { uri: string },
  address: String,
  goToOtherWindow: Function,
};
class SideMenu extends Component<Props> {
  navigateToScreen = route => () => {
    const {
      navigation: { dispatch },
    } = this.props;
    const navigateAction = NavigationActions.navigate({
      routeName: route,
    });
    dispatch(navigateAction);
  };

  handleDashboard = () => {
    const {
      navigation: { closeDrawer },
    } = this.props;
    closeDrawer();
    this.navigateToScreen('Dashboard')();
  };

  handleSend = () => {
    const {
      navigation: { closeDrawer },
      setModalSend,
    } = this.props;
    closeDrawer();
    this.navigateToScreen('Dashboard')();
    setModalSend(true);
  };

  handleReceive = () => {
    const {
      navigation: { closeDrawer },
      setModalReceive,
    } = this.props;
    closeDrawer();
    this.navigateToScreen('Dashboard')();
    setModalReceive(true);
  };

  handleSetup = () => {
    const {
      navigation: { closeDrawer },
    } = this.props;

    closeDrawer();
    this.navigateToScreen('Settings')();
  };

  handleTransactions = () => {
    const {
      navigation: { closeDrawer },
    } = this.props;

    closeDrawer();
    this.navigateToScreen('Transactions')();
  };

  handleNotifications = () => {
    const {
      navigation: { closeDrawer },
      setVisibleNotify,
    } = this.props;
    closeDrawer();
    setVisibleNotify(true);
    this.navigateToScreen('Dashboard')();
  };

  handleProfiles = () => {
    const {
      navigation: { closeDrawer },
    } = this.props;

    closeDrawer();
    this.navigateToScreen('Profile')();
  };

  renderBottomButtonForSmall = () => {
    if (heightWindow <= 600) {
      return (
        <View style={styles.links}>
          <TouchableOpacity style={styles.link} onPress={this.handleSetup}>
            <Image style={styles.icon} source={setup} />
            <Text style={styles.title}>Settings</Text>
          </TouchableOpacity>
        </View>
      );
    }
    return null;
  };

  renderBottomButton = () => {
    if (heightWindow > 600) {
      return (
        <View style={styles.linksBottom}>
          <TouchableOpacity style={styles.linkBottom} onPress={this.handleSetup}>
            <Image style={styles.icon} source={setup} />
            <Text style={styles.title}>Settings</Text>
          </TouchableOpacity>
        </View>
      );
    }
    return null;
  };

  shareOnlyAddress = () => {
    const { address, goToOtherWindow } = this.props;

    if (address) {
      goToOtherWindow(true);

      const shareImageBase64 = {
        title: 'Lapo address',
        message: address,
        failOnCancel: false,
      };

      Share.open(shareImageBase64).catch();
    }
  }

  render() {
    const { address, username, avatarSource } = this.props;
    return (
      <View style={styles.container}>
        <ScrollView>
          <SafeAreaView>
            <View style={styles.header}>
              {(username !== '' || avatarSource.uri !== '') && (
                <View style={styles.headeColumns}>
                  <Image style={styles.foto} source={avatarSource} />
                  <Text style={styles.headerTitle}>{username}</Text>
                </View>
              )}
              <TouchableOpacity onPress={this.shareOnlyAddress}>
                {address !== '' && <Text style={styles.headerSub}>{address}</Text>}
              </TouchableOpacity>
            </View>

            <View style={styles.links}>
              <TouchableOpacity style={styles.link} onPress={this.handleDashboard}>
                <Image style={styles.icon} source={dashboard} />
                <Text style={styles.title}>Dashboard</Text>
              </TouchableOpacity>
              <TouchableOpacity style={styles.link} onPress={this.handleSend}>
                <Image style={styles.icon} source={send} />
                <Text style={styles.title}>Send LAX</Text>
              </TouchableOpacity>
              <TouchableOpacity style={styles.link} onPress={this.handleReceive}>
                <Image style={styles.icon} source={recieve} />
                <Text style={styles.title}>Receive LAX</Text>
              </TouchableOpacity>
              <TouchableOpacity style={styles.link} onPress={this.handleTransactions}>
                <Image style={styles.icon} source={trans} />
                <Text style={styles.title}>Transactions</Text>
              </TouchableOpacity>
              <TouchableOpacity style={styles.link} onPress={this.handleNotifications}>
                <Image style={styles.icon} source={notify} />
                <Text style={styles.title}>Notifications</Text>
              </TouchableOpacity>

              <TouchableOpacity style={styles.link} onPress={this.handleProfiles}>
                <Image style={styles.icon} source={profile} />
                <Text style={styles.title}>Profile</Text>
              </TouchableOpacity>

              {this.renderBottomButtonForSmall()}
            </View>
          </SafeAreaView>
        </ScrollView>
        {this.renderBottomButton()}
      </View>
    );
  }
}

const mapStateToProps = (state: any): Object => ({
  address: state.wallet.address,
  avatarSource: state.settings.avatar,
  username: state.settings.username,
});

const mapDispatchToProps = (dispatch: Dispatch<*>) => bindActionCreators(
  {
    setModalSend: setModalSendAction,
    setModalReceive: setModalReceiveAction,
    setPaperWordsConfirm: setPaperWordsConfirmAction,
    setPins: setPinsAction,
    setConfirmPins: setConfirmPinsAction,
    setVisibleNotify: actionNotify.setVisibleNotify,
    goToOtherWindow: goToOtherWindowAction,
  },
  dispatch,
);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(SideMenu);
