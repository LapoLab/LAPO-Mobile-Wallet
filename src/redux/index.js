import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import storage from 'redux-persist/lib/storage';
import { persistStore, persistReducer } from 'redux-persist';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import rootReducer from './rootReducer';
import rootSagas from './rootSaga';

const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
  blacklist: [
    'modals',
    'dashboard',
    'paperWords',
    'chooseContact',
    'send',
    'transactionsDetails',
    'alert',
    'pinsRecovery',
  ],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const sagaMiddleware = createSagaMiddleware();
// eslint-disable-next-line no-undef
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = createStore(
  persistedReducer,
  composeEnhancers(applyMiddleware(sagaMiddleware)),
);
sagaMiddleware.run(rootSagas);

export const persistor = persistStore(store);
