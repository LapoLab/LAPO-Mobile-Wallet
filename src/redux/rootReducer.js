import { combineReducers } from 'redux';

import wallet from '../commons/reducers/wallet';
import pins from '../commons/reducers/pins';
import pinsRecovery from '../commons/reducers/pinsRecovery';
import modals from '../commons/reducers/modals';
import settings from '../commons/reducers/settings';
import notify from '../modules/Notify/reducers';
import paperConfirm from '../modules/PaperWordsConfirm/reducer';
import dashboard from '../modules/Dashboard/reducers';
import paperWords from '../modules/PaperWords/reducers';
import chooseContact from '../modules/ChooseContact/reducers';
import send from '../modules/Send/reducers';
import transactionsDetails from '../modules/TransactionDetails/reducer';
import alert from '../modules/Alert/reducers';

export default combineReducers({
  wallet,
  pins,
  pinsRecovery,
  notify,
  modals,
  settings,
  paperConfirm,
  dashboard,
  paperWords,
  chooseContact,
  send,
  transactionsDetails,
  alert,
});
