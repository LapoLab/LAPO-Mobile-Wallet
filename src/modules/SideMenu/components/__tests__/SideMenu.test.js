// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../../../redux';
import SideMenu from '../index';

describe('SideMenu page', () => {
  test('renders snapshots', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <SideMenu />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
