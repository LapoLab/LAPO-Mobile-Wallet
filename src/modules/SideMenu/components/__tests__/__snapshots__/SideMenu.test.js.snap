// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu page renders snapshots 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#160A3A",
      "flex": 1,
      "padding": 23,
    }
  }
>
  <RCTScrollView>
    <View>
      <RCTSafeAreaView>
        <View
          style={
            Object {
              "flex": 1,
              "marginBottom": 20,
              "marginTop": 0,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "marginBottom": 40,
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/dashboard.png",
                }
              }
              style={
                Object {
                  "height": 24,
                  "resizeMode": "contain",
                  "width": 54,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                }
              }
            >
              Dashboard
            </Text>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "marginBottom": 40,
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/send.png",
                }
              }
              style={
                Object {
                  "height": 24,
                  "resizeMode": "contain",
                  "width": 54,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                }
              }
            >
              Send LAX
            </Text>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "marginBottom": 40,
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/recieve.png",
                }
              }
              style={
                Object {
                  "height": 24,
                  "resizeMode": "contain",
                  "width": 54,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                }
              }
            >
              Receive LAX
            </Text>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "marginBottom": 40,
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/trans.png",
                }
              }
              style={
                Object {
                  "height": 24,
                  "resizeMode": "contain",
                  "width": 54,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                }
              }
            >
              Transactions
            </Text>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "marginBottom": 40,
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/notify.png",
                }
              }
              style={
                Object {
                  "height": 24,
                  "resizeMode": "contain",
                  "width": 54,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                }
              }
            >
              Notifications
            </Text>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "marginBottom": 40,
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/profile.png",
                }
              }
              style={
                Object {
                  "height": 24,
                  "resizeMode": "contain",
                  "width": 54,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                }
              }
            >
              Profile
            </Text>
          </View>
        </View>
      </RCTSafeAreaView>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "column",
        "height": 50,
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/setup.png",
          }
        }
        style={
          Object {
            "height": 24,
            "resizeMode": "contain",
            "width": 54,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 18,
          }
        }
      >
        Settings
      </Text>
    </View>
  </View>
</View>
`;
